cloudsync/event.py

```diff
@@ -197,17 +197,20 @@ def _do_walk_if_needed(self):
             log.debug("walking all %s/%s-%s files as events, because no working cursor on startup",
                       self.provider.name, self._root_path, self._root_oid)
             self._queue = []
-            self._do_walk_oid(self._root_oid)
-            self.state.storage_update_data(self._walk_tag, time.time())
-            self.need_walk = False
+            if self._do_walk_oid(self._root_oid):
+                self.state.storage_update_data(self._walk_tag, time.time())
+                self.need_walk = False
 
-    def _do_walk_oid(self, oid):
+    def _do_walk_oid(self, oid) -> bool:
         try:
             if oid:
                 for event in self.provider.walk_oid(oid):
                     self._process_event(event, from_walk=True)
+                    if self.stopped:
```

Comment:
I think the `stopped` check belongs before the `_process_event()` call. If we move the check and stop happens during the call to `walk_oid()` we can prevent the first event, and if it happens during the last event, it will allow the storage to be updated to show that the walk completed, which it did, and which should hopefully be a pretty quick operation.

Reply:
good point.